docs/enhancements/sep-tarball-client.md

```diff
@@ -0,0 +1,189 @@
+# Sonobuoy Tarball Client
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Objectives](#objectives)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [User Stories](#user-stories)
+* [Unresolved Questions](#unresolved-questions)
+
+## Summary
+
+Sonobuoy delivers results in a gzipped tar archive. The structure of the tar is
+[documented here][tar structure] but programmatic access is up to every consumer
+to implement. The structure of the tarball can change between versions and
+consumer implementations become fragile.
+
+This proposal aims to describe a client library to help automate the analysis of
+a Sonobuoy result artifact.
+
+## Objectives
+
+Provide a client library for easy access to data found in Sonobuoy results
+across sonobuoy and Kubernetes versions.
+
+### Goals
+
+ - Support all tarball versions starting at v0.9.0.
+ - Provide basic access to kubernetes resources and plugin-generated files.
+ - Provide good access to Sonobuoy-generated data.
+
+### Non-Goals
+
+- Manage Kubernetes resource versioning.
+- Customize the client library for every plugin written.
+- There will be no support for tarballs before Sonobuoy version 0.9.0.
+
+## Proposal
+
+```go
+type Result interface {
+    // NamespacedResources returns a list of resources (the entire collection
+    // can be seen in the sonobuoy config) as `*runtime.Unknown`s to have this
+    // client avoid kubernetes //resource versions.
+    NamespacedResources() []*runtime.Unknown
```

Comment:
You'll want to be able to walk anything that is a list.  

Perhaps instead of outlining the details in the doc which is fraught with perl, include w/ the PR a coded set of interfaces that are light, b/c it will force you to experience the UX.

Reply:
That was the idea behind this whole section...design it based on how I'd use it.

I don't really understand why I'd want to walk this list. I would expect a `for` loop to be sufficient, but I must be missing something. Can you explain why I need to walk every list?

Reply:
Folks may want to prefilter the data, and we need a means for them to iterate over the namespace-scoped resources for each resource kind, b/c you can't load them all from disk in 1 go without it hogging the machine, and they may want to skip sections based on 'kind'.

Reply:
Discussed offline. Big clusters are big.